### ISSUE TITLE
`create-webiny-project` and other DX improvements

### ISSUE THREAD

**Issue by Pavel910** [webiny/webiny-js#758]
## Background
While working on #730, we discovered many small but very annoying DX issues and realized we need to change how we handle API services (this includes both YAML template, code location, building, TS configs, etc) and overall project setup.

## Goals
The goals is always the same - to make developer experience the best it can be! We want to provide our users a complete project setup that has all the necessary tools configured and ready to go. We also want to provide scaffolding tools so users don't waste time on creating new services, functions, packages. Moreover, we want our setup to be easy to manage and upgrade when we (Webiny) release breaking changes in React apps (this will be possible with #714).

## What we need to do
So to keep things straight to the point, here is a list of tasks we need to finish before we can officially release the scaffolding tool:

- [x] create a `create-webiny-project` package that will be used to setup the Webiny project, and replace the current `webiny create` command (#764).
- [x] the project must mimic our own Webiny monorepo, and be ready for development of custom packages with all TS and babel configs (currently, we do not have this in our project setup and a user is in a world of pain if he wants to develop a new package and use TS)
- [x] API services must be generated as actual code in the selected folder, to allow proper webpack bundling and more importantly - `watching` and `redeployment` during development. I have a working POC of this in another repository. This is also a prerequisite for proper scaffolding (that's why #733 is put on hold).
- [x] React apps must be setup using #714 (for this I also have a working POC)
- [x] separates `build` stage from `deploy` stage in `serverless.yml` template, which will make the template significantly smaller and easier to read, and allow us to implement the `watch` system
- [x] Create an `apollo-service-handler` to use with our `@webiny/http-handler` package @doitadrian 
- [x] add support for multiple build stages (simple array of objects), each stage can have its own `root` path
- [x] `api` and `apps` are no longer built-in CLI concepts. You simply run `webiny deploy api` and CLI will know it needs to read `serverless.yml` from `./api` folder. This gives us the freedom of defining as many APIs and app groups as we want. We also no longer force developers on any specific folder naming.
- [x] a folder containing `serverless.yml` can contain `webiny.config.js` which points to hooks (simple paths where we look for other `webiny.config.js` files, and execute hooks within them)
- [x] separate `site` into separate packages and serverless resources for deployment. This will make the whole logic much more  decoupled, easier to read and easier to deploy.
- [x] `@webiny/cli` introduces a `run` command to execute commands defined in `webiny.config.js` files
- [x] support resource definition via JS
- [ ] @doitadrian will handle app deployment to S3 and CDN setup (to remove the need for Lambda and API Gateway to serve static assets)

Besides handling all of the aforementioned issues, this upgrade will take care of other annoyances reported by different users so I expect this release to be a major overall improvement for us and our community.

## Why all this? Why is it so complicated?
Proper `monorepo` setup requires many different tools to work together: TS, babel, webpack, React,.... There are many details involved in setting up a proper monorepo project which can handle big projects. Webiny repo currenty has 60+ npm packages; it didn't happen overnight, so along the way we tested and tried many approaches. At this point we can say that our setup works, and we want to share it and help you get going with your project instead of fighting dev tooling setup.

## What we have so far
While experimenting and building a POC, I've reached a point where I can say this `works` and can actually be polished to become the next project template:

![image](https://user-images.githubusercontent.com/3920893/76758374-dcc06c00-6789-11ea-962b-f3590fe21bc6.png)

How API watch mechanism works:
![CLIWatch](https://user-images.githubusercontent.com/3920893/76759369-cf0be600-678b-11ea-8018-5caf6fb963dd.gif)

**Comment by doitadrian** [comment]
1. From my experience, the "waiting" functionality wasn’t super needed, because you would get used to the fact that each save = deploy very quickly. It really has become second nature very quickly.

2. ~It would be cool to have webpack stop creating a build, if in the process another change comes to the code from the dev.~ Not possible for now.

3. I was also thinking about the same for the component-deployment-process: stop deployment on a code-change. I am aware that you might end up with half-deployed service, but this is development. It shouldn't matter much in my opinion.

**Comment by Pavel910** [comment]
We'll see if this delayed deploy is something that will make sense once we start working on page builder and those bigger apps, as they could build longer (even on incremental builds) and maybe they will last longer than those 3 seconds - we'll see as we go.

You can't abort webpack build, it handles itself, so whenever you change something it will rebuild itself. Once it is done building, it calls the `deploy` callback which then waits for 3 seconds for any subsequent `deploy` calls, and if it is not "interrupted" it will deploy the resource.

I couldn't abort an `in-progress` deploy even if I wanted, since that implies killing the node process. Since deploys take seconds, I don't see a problem with it. One thing we can do is maybe queue the deploys, so if a change comes in the middle of current deploy, it will let it finish and then process to the new deployment cycle.
